### PR TITLE
PF-1728: add the metadata display in the text/json

### PR DESCRIPTION
**Pull request by ginaygoogle**
image
update one metadata

Screen Shot 2022-06-16 at 7 58 20 PM
add multiple entries of metadata.

### CODE REVIEW

File: src/test/java/unit/GcpNotebookControlled.java [hunk 1]
```diff
@@ -211,6 +213,12 @@ void overrideLocationAndInstanceId() throws IOException {
     // stored in or accessible via Workspace Manager.
     assertEquals(newName, updatedNotebook.name, "create output matches name");
     assertEquals(newDescription, updatedNotebook.description, "create output matches description");
+    String[] metadatas = newMetadata.split(",");
```
Comment (yuhuyoyo) [inline src/test/java/unit/GcpNotebookControlled.java:216]: when you use a map, the correct format is:
--new-metadata=newFoo=newBar --new-metadata=newFoo2=newBar2. you shouldn't need to do split yourself. See usage example: https://picocli.info/#_maps

Comment (jaycarlton) [reply src/test/java/unit/GcpNotebookControlled.java:216]: If you use the `split` argument, you can also do `--new-metadata=foo=bar,baz=quux`, but yes, you don't need to call `String#split`.

File: src/test/java/unit/GcpNotebookControlled.java [hunk 2]
```diff
@@ -204,13 +207,19 @@ void overrideLocationAndInstanceId() throws IOException {
...
-            "--new-metadata=" + newMetadata);
+            "--new-metadata=" + newMetadata1 + "," + newMetadata2);
 
     // check that the properties match
     // the metadata supports multiple entries, we can't assert on the metadata because it's not
     // stored in or accessible via Workspace Manager.
     assertEquals(newName, updatedNotebook.name, "create output matches name");
     assertEquals(newDescription, updatedNotebook.description, "create output matches description");
+    assertTrue(
+        updatedNotebook.metadata.toString().contains(newMetadata1),
```
Comment (yuhuyoyo) [inline src/test/java/unit/GcpNotebookControlled.java:218]: i think i prefer `assertEquals(metadata1, updatedNotebook.metadata.get(NewMetadata1))` than converting map to string.

Comment (jaycarlton) [reply src/test/java/unit/GcpNotebookControlled.java:218]: ➕

File: src/test/java/unit/GcpNotebookControlled.java [hunk 3]
```diff
@@ -206,19 +207,19 @@ void overrideLocationAndInstanceId() throws IOException {
             "--name=" + name,
             "--new-name=" + newName,
             "--new-description=" + newDescription,
-            "--new-metadata=" + newMetadata);
+            "--new-metadata=" + newMetadata1 + "," + newMetadata2);
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply src/test/java/unit/GcpNotebookControlled.java:210]: By adding the `split` option, now the user can do it either way. We should probably do this for the other key-value options.

File: src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java [hunk 4]
```diff
@@ -62,6 +66,10 @@ public void print(String prefix) {
     OUT.println(prefix + "Instance id:       " + instanceId);
     OUT.println(prefix + "Instance name: " + (instanceName == null ? "(undefined)" : instanceName));
     OUT.println(prefix + "State:         " + (state == null ? "(undefined)" : state));
+    OUT.println(prefix + "Metadata:");
+    for (Map.Entry<String, String> entry : metadata.entrySet()) {
+      OUT.println("   " + entry.getKey() + ":" + entry.getValue());
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java:71]: Put a space after the colon like ": ". Also try a tab to see what looks better (":\t").

File: src/test/java/unit/GcpNotebookControlled.java [hunk 5]
```diff
@@ -48,6 +48,8 @@ void listDescribeReflectCreateDelete() throws IOException {
 
     // check that the name and notebook name match
     assertEquals(name, createdNotebook.name, "create output matches name");
+    assertTrue(
+        createdNotebook.metadata.toString().contains("foo=bar"), "create output matches metadata");
```
Comment (jaycarlton) [inline src/test/java/unit/GcpNotebookControlled.java:52]: You don't need to convert the map to a string. Do a `get()` on it like `assertEquals("bar", createdNotebook.metadata.get("foo"))`

**Review by jaycarlton — APPROVED** [review]

File: src/test/java/unit/GcpNotebookControlled.java [hunk 6]
```diff
@@ -194,7 +195,8 @@ void overrideLocationAndInstanceId() throws IOException {
     // new key-value pair will be appended, existing key-value pair will be updated.
     String newName = "NewOverrideLocationAndInstanceId";
     String newDescription = "\"new override default location and instance id\"";
-    String newMetadata = "NewMetadata1=metadata1,NewMetadata2=metadata2";
+    String newMetadata1 = "NewMetadata1=metadata1";
```
Comment (jaycarlton) [inline src/test/java/unit/GcpNotebookControlled.java:198]: I'd declare `String newKey1 = "NewMetadata1;`, etc, and then do `String newEntry1 = newKey1 + "=" + newValue1;`. Then when you go to do your assertions, you don't have to split to get the expected value for `newValue1`.

File: src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java [hunk 7]
```diff
@@ -62,6 +66,10 @@ public void print(String prefix) {
     OUT.println(prefix + "Instance id:       " + instanceId);
     OUT.println(prefix + "Instance name: " + (instanceName == null ? "(undefined)" : instanceName));
     OUT.println(prefix + "State:         " + (state == null ? "(undefined)" : state));
+    OUT.println(prefix + "Metadata:");
+    for (Map.Entry<String, String> entry : metadata.entrySet()) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java:70]: You can also do 
```
metadata.forEach((key, value) -> OUT.println("   " + key + ": " + value));
```
I think it's cleaner than peeling off the entrySet when you don't have to.

File: src/test/java/unit/GcpNotebookControlled.java (hunk 6 above)
Comment (ginaygoogle) [reply src/test/java/unit/GcpNotebookControlled.java:198]: I was also thinking that way to separate the key and value. but the concern will be we should have 4 more variables... I am not sure comparing the more redundant variables and a split sentence, which one will make the code better.

File: src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java (hunk 7 above)
Comment (ginaygoogle) [reply src/main/java/bio/terra/cli/serialization/userfacing/resource/UFGcpNotebook.java:70]: great!

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**